elastic/kibana
# feat(slo): allow configuration of advanced settings from UI

Resolves https://github.com/elastic/kibana/issues/200041

### 🐎 Summary

This PR adds an **advanced settings** section at the end of the objective section in the SLO form. This section can be used to configure the following settings fields: **syncField**, **syncDelay**, **frequency** and **preventInitialBackfill**. 

**syncField** is a new settings. The other fields were already defined at the API level.
Except for **preventInitialBackfill**, the other fields were not available on the form.

**syncField** is optional, therefore current usage of the create/update APIs won't break if this field is not specified. A **null** value is used to specifically unset the field when updating an SLO.

![image](https://github.com/user-attachments/assets/2ca8fe84-dd63-4912-8145-2a72443091e8)

I've also taken the opportunity to close https://github.com/elastic/kibana/issues/194957 by fixing the `index` row label state when the form is invalid.

![image](https://github.com/user-attachments/assets/216aeb9d-37a2-4344-ae30-32f6a42033f2)




> [!NOTE]
> While implementing this feature, I've noticed some bugs in the form state when changing the DataView / selected index. The dependant fields like **groupBy** and the new **syncField** were not cleared from the previously selected values, but the UI was hidding them.

The form was keeping the previous selected fields in its state, while hiding them from the combobox selection. This could lead to subtle bugs where user would edit an existing SLO, changing the DataView and hitting save, thinking the groupBy/syncField were cleared.

I've fixed this in this PR as well.




### 🧬 Testing

- Creating an SLO should:
  - Use the syncField, syncDelay, Frequency provided values to build the rollup transform
  - Use default values if not provided
- Editing an SLO should
  -  Use the syncField, syncDelay, Frequency provided values to build the rollup transform
  - Keep the previous settings if not overridden

## Release Note

- Adds **syncField**, **syncDelay** and **frequency** settings in the API and SLO form. Allow customer to fine tune their SLO settings directly from the UI.
